py/datastore_service.py

```diff
@@ -40,35 +45,33 @@ def serve_forever(self):
     self.server_close()
 
   def make_thread_locals(self):
-    credentials = (os.environ['POSTGRES_DB'], os.environ['POSTGRES_USER'], os.environ['POSTGRES_HOST'], 
-                   os.environ['POSTGRES_PASSWORD'], os.environ['POSTGRES_PORT'])
+    # connect to the db
     try:
-      sql_conn = psycopg2.connect("dbname='%s' user='%s' host='%s' password='%s' port='%s'" % credentials)
+      self.credentials = (os.environ['POSTGRES_DB'], os.environ['POSTGRES_USER'], os.environ['POSTGRES_HOST'], 
+                          os.environ['POSTGRES_PASSWORD'], os.environ['POSTGRES_PORT'])
+      sql_conn = psycopg2.connect("dbname='%s' user='%s' host='%s' password='%s' port='%s'" % self.credentials)
+      sql_conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+      setattr(thread_local, 'sql_conn', sql_conn)
```

Comment:
this is the important part. we hang values off of the `thread_local` object, which is a different object for every thread that accesses it. this is how we make sure to get a real new connection for every thread